### DISCUSSION TITLE
Is there a way to pass nonce to bypass CSP with devtools script?

### DISCUSSION THREAD

**Discussion by uladzimirdev** [thheller/shadow-cljs#1199] [General]
I need to add `nonce` to the script somehow

<img width="1131" alt="image" src="https://github.com/user-attachments/assets/0cbe9af5-ae1b-4715-b861-ef077cf11806">

**Comment by thheller** [comment]
Need more information to understand what you are asking. shadow-cljs doesn't generate any script tags or inline styles, so I'm not sure what is going on there.

**Reply by uladzimirdev** [reply]
I'm sorry, I'm very new to cljs, we use your package with webpack live reload and I just checked devtools and the origin of the problem was 
<img width="1192" alt="image" src="https://github.com/user-attachments/assets/5d8fca1a-26e8-4aac-95f7-4eae90cad9de">

the very top method calls `elem.innerHTML = goog.html.SafeHtml.unwrapTrustedHTML(html);`

and html looks like this 

```
{
  "privateDoNotAccessOrElseSafeHtmlWrappedValue_": "<div style=\"transition:opacity 1s linear;-moz-transition:opacity 1s linear;\"></div>",
  "implementsGoogStringTypedString": true
}
```

it says nothing to me, maybe the issue is not in shadow-cljs at all

**Reply by uladzimirdev** [reply]
@thheller the only real thing I found was a require statement

```
require("./shadow.cljs.devtools.client.browser.js");
```

if I comment it out, the error disappears

**Reply by thheller** [reply]
What is doing the live reloading? shadow-cljs or webpack? Looks like you are using a `:npm-module` build? To build what exactly? I still do not quite understand what you are doing.

**Reply by uladzimirdev** [reply]
sorry, we add event listener and webpack is doing live-reloading. we're using npm module

here is the problematic line https://github.com/metabase/metabase/blob/master/src/metabase/util/devtools.cljc#L7

**Reply by thheller** [reply]
When you are using webpack to do the reloading, then you do not really need the shadow-cljs devtools. Unless you are still using the REPL? You could otherwise set `:devtools {:enabled false}` in the build config, but that manual require then needs to be removed. Unsure why it is there in the first place.

I can see about removing that inline style thing. Didn't even know it tried to do that. It is from some very very old code that I haven't looked at in a decade probably.

**Reply by thheller** [reply]
Try `2.28.12`. That reference is gone.

**Reply by uladzimirdev** [reply]
~Thanks! I've tried 2.28.12, but the message still exists.~

It worked! thank you very much 🎉 

UPD I needed to update deps.edn as well, sorry for confusion!



Regarding why it's done like this, I'll investigate further. In perfect world I'd love to build ESM as well, so I'll investigate further
